src/Custom/Embeddings/OpenAIEmbedding.cs

```diff
@@ -121,6 +121,7 @@ private static ReadOnlyMemory<float> ConvertToVectorOfFloats(BinaryData binaryDa
         OperationStatus status = Base64.DecodeFromUtf8(base64, bytes.AsSpan(), out int bytesConsumed, out int bytesWritten);
         if (status != OperationStatus.Done || bytesWritten % sizeof(float) != 0)
         {
+            ArrayPool<byte>.Shared.Return(bytes);
```

Comment:
Can we please refactor the approach to move this to a `finally` block?  We're likely to run into additional missed corner cases over time and will already leak in the case of an unexpected exception.

Reply:
done!